### PR TITLE
Add AWS EMR and Athena components

### PR DESCRIPTION
**Pull request by Jeffwan**
1. Add AWS EMR and Athena components
2. Add example to use EMR to run Spark pipelines.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1286)

### CODE REVIEW

File: components/aws/athena/query/component.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# limitations under the License.
+
+name: Athena - Query
+  A Kubeflow Pipeline component to submit a query to Amazon Web Services Athena
+  service and dump outputs to AWS S3.
+
+inputs:
+  - {name: region,        description: 'The Athena region in which to handle the request.'}
+  - {name: database,      description: 'The name of the database.'}
+  - {name: query,         description: 'The SQL query statements to be executed in Athena.}
```
Comment (Ark-kun) [inline components/aws/athena/query/component.yaml]: The tests a failing - missing ending quote.
```suggestion
  - {name: query,         description: 'The SQL query statements to be executed in Athena.'}
```

File: components/aws/athena/query/src/query.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          'OutputLocation': output,
+          }
+      )
+
+  execution_id = response['QueryExecutionId']
+  logging.info('Execution ID: %s', execution_id)
+
+  # Athena query is aync call, we need to fetch results and wait for execution
+  state = 'RUNNING'
+  max_execution = 5 # TODO: this should be an optional parameter from users. or use timeout
```
Comment (animeshsingh) [inline components/aws/athena/query/src/query.py]: Yes - can be a default param in input

File: components/aws/athena/query/component.yaml (hunk 1 above)
Comment (Jeffwan) [reply components/aws/athena/query/component.yaml]: Thanks! I miss this quota in latest commit.

File: components/aws/athena/query/src/query.py (hunk 2 above)
Comment (Jeffwan) [reply components/aws/athena/query/src/query.py]: Thanks. I will give to our users and see how do they like it.  Change will be made and expose to users later either with max_execution or max_waiting_time.

**Pull request merged**